python/ray/train/mosaic/_mosaic_utils.py

```diff
@@ -0,0 +1,225 @@
+        session.report(self.data)
+
+    def epoch_checkpoint(self, state: State, logger: Logger) -> None:
+        del logger  # unused
+        session.report(self.data)
+
+
+class RayTrainReportCallback(CheckpointSaver):
+    """A callback that wraps Composer's ``CheckpointSaver``.
+
+    This class is used to wrap each Composer ``CheckpointSaver`` to be used in Composer
+    trainer. The main role of this callback is to report the paths of the checkpoints
+    saved by the Composer ``CheckpointSaver`` it wraps. In addition, when the training
+    ends, (either with or without exception) the last checkpoint and the list of all
+    the checkpoints that have been saved and the list of Composer InMemoryLogger are
+    reported.
+
+    Example:
+        .. code-block:: python
+            # create a MosaicTrainer
+            mosaic_trainer =  MosaicTrainer(
+                    trainer_init_per_worker=trainer_init_per_worker,
+                    datasets={"train": train_dataset},
+                    trainer_init_config=trainer_init_config,
+                    scaling_config=scaling_config
+                )
+            result = mosaic_trainer.fit()
+
+            chkpt_dict = result.checkpoint.to_dict()
+
+            in_memory_logger = chkpt_dict["in_memory_logger"]
+            last_checkpoint = chkpt_dict["last_checkpoint"]
+            all_checkpoints = chkpt_dict["all_checkpoints"]
+
+    Args:
+        in_memory_logger: The list of Composer InMemoryLogger that would be used in
+            Composer trainer initialization.
+        checkpoint_saver: A Composer ``CheckpointSaver`` that the callback will wrap.
+            If this argument is provided, then the parent class is initialized with the
+            passed in ``CheckpointSaver`` object's attributes. Otherwise, the parent
+            class is initialized with args provided.
+        args: Arguments for initializing a Composer ``CheckpointSaver`` object.
+    """
+
+    def __init__(
+        self, in_memory_logger, checkpoint_saver: CheckpointSaver = None, **args
+    ):
+        self.in_memory_logger = in_memory_logger
+        self.last_checkpoint = None
+        self.checkpoint_count = 0
+
+        if checkpoint_saver:
+            super(RayTrainReportCallback, self).__init__(
+                checkpoint_saver.folder,
+                checkpoint_saver.filename,
+                checkpoint_saver.artifact_name,
+                checkpoint_saver.latest_filename,
+                checkpoint_saver.latest_artifact_name,
+                checkpoint_saver.save_interval,
+                overwrite=checkpoint_saver.overwrite,
+                num_checkpoints_to_keep=checkpoint_saver.num_checkpoints_to_keep,
+                weights_only=checkpoint_saver.weights_only,
+                **args
+            )
+        else:
+            super(RayTrainReportCallback, self).__init__(**args)
+
+    def close(self, state: State, logger: Logger) -> None:
+        del logger  # unused
```

Comment:
is this necessary? I don't think we should explicitly delete variables

Reply:
This is the general practice shown in the Composer library, so I followed their practice. This can be removed if deemed extra

Reply:
If this is the practice they are using, then that's fine with me, though I'd add a comment somewhere (could be just at the top of the file/before the class) that this is their practice.